parser_utils.py

```diff
@@ -0,0 +1,331 @@
+        self.op = op
+
+    def __repr__(self):
+        return "{} {}".format(self.coeff, self.op)
+
+
+class Measure(Op):
+    """Class for representing the 'measure' instruction."""
+    def __init__(self, wires):
+        super().__init__(Ops.MEASURE, "measure", params=[], wires=wires)
+
+    def __repr__(self):
+        return f"{self.name} {format_wires(self.wires[0])} -> {format_wires(self.wires[1])};"
+
+
+class Barrier(Op):
+    """Class for representing the 'barrier' keyword."""
+    def __init__(self, wires):
+        super().__init__(Ops.BARRIER, "barrier", params=[], wires=wires)
+
+
+class EqualityCondition:
+    """Class for representing the classical equality expression from the specification.
+
+    For example, as appears in ``if(c==5) CX q[0],q[1];``.
+    """
+    def __init__(self, id_, integer):
+        self.id = id_
+        self.integer = integer
+
+    def __repr__(self):
+        return f"{self.id} == {self.integer}"
+
+
+class ConditionalOp:
+    """Class for representing the classical conditional expression from the specification.
+
+    For example, as appears in ``barrier q[0],q[1];``."""
+    def __init__(self, condition, op):
+        self.condition = condition
+        self.op = op
+
+    def __repr__(self):
+        return f"if ({self.condition}) {self.op}"
+
+
+class Declaration:
+    """Base class for representing the declaration of an object that is supported in the grammar."""
+    def __init__(self, decl_type, opaque=False, **kwargs):
+        self.decl_type = decl_type
+        self.name = decl_type.name.lower()
+        self.opaque = opaque
+        self.kwargs = kwargs
+        self.goplist = []
+
+    def declaration_str(self):
+        return "{} declaration, kwargs={}".format(self.name, self.kwargs)
+
+    def __repr__(self):
+        if self.opaque or not self.goplist:
+            return self.declaration_str()
+
+        output = [self.declaration_str(), "{"]
+
+        for op in self.goplist:
+            # TODO: improve this so tensor products print like "x a, y b;"
+            # (they are not treated same as goplist)
+            output.append(f"    {op}")
+
+        output.append("}")
+        return "\n".join(output)
+
+
+class ClassicalRegister(Declaration):
+    """Class for representing the declaration of classical registers.
+
+    For example, as in ``creg c[5];``"""
+    def __init__(self, name, size):
+        super().__init__(decl_type=Declarations.CREG, id_=name, size=size)
+
+    def __repr__(self):
+        return f"creg {self.kwargs['id_']}[{self.kwargs['size']}];"
+
+
+class QuantumRegister(Declaration):
+    """Class for representing the declaration of quantum registers.
+
+    For example, as in ``qreg q[5];``"""
+    def __init__(self, name, size):
+        super().__init__(decl_type=Declarations.QREG, id_=name, size=size)
+
+    def __repr__(self):
+        return f"qreg {self.kwargs['id_']}[{self.kwargs['size']}];"
+
+
+class GateDeclaration(Declaration):
```

Comment:
Removed `QDeclaration`, which wasn't doing anything useful

Reply:
`QDeclaration` comes from the language spec, but I agree, it's not very clear how it is useful.